### PR TITLE
[CORDA-2084] EdDSA, SPHINCS-256 and RSA PKCS#1 are deterministic, no RNG required.

### PR DESCRIPTION
**Pull request by kchalkias**
In practice, out of the 5 supported signature schemes, only ECDSA R1/K1 is not deterministic. For the rest, no `newSecureRandom` is required when signing.

### CODE REVIEW

File: core/src/test/kotlin/net/corda/core/crypto/CryptoUtilsTest.kt [hunk 1]
```diff
@@ -786,110 +795,131 @@ class CryptoUtilsTest {
...
         assertEquals("DL9yJfiNGqteRrKPjGUkRQkeqzuQ4kwcYQWMCi5YKuUHrk", keyPairBiggerThan256bitsV2.public.toStringShort())
-        val keyPairBiggerThan512bits = Crypto.deriveKeyPairFromEntropy(Crypto.ECDSA_SECP256K1_SHA256, BigInteger("2").pow(514).minus(BigInteger.TEN))
+        val keyPairBiggerThan512bits = Crypto.deriveKeyPairFromEntropy(ECDSA_SECP256K1_SHA256, BigInteger("2").pow(514).minus(BigInteger.TEN))
         assertEquals("DL3Wr5EQGrMTaKBy5XMvG8rvSfKX1AYZLCRU8kixGbxt1E", keyPairBiggerThan512bits.public.toStringShort())
-        val keyPairBiggerThan258bits = Crypto.deriveKeyPairFromEntropy(Crypto.ECDSA_SECP256K1_SHA256, BigInteger("2").pow(259).plus(BigInteger.ONE))
+        val keyPairBiggerThan258bits = Crypto.deriveKeyPairFromEntropy(ECDSA_SECP256K1_SHA256, BigInteger("2").pow(259).plus(BigInteger.ONE))
         assertEquals("DL7NbssqvuuJ4cqFkkaVYu9j1MsVswESGgCfbqBS9ULwuM", keyPairBiggerThan258bits.public.toStringShort())
     }
+    
+    @Test
```
Comment (IgorNitto) [inline core/src/test/kotlin/net/corda/core/crypto/CryptoUtilsTest.kt]: Lots of minor changes here do not seem related to testing deterministic signature schemes, is this deliberate?

Comment (kchalkias) [reply core/src/test/kotlin/net/corda/core/crypto/CryptoUtilsTest.kt]: yeah, I took the opportunity to reduce long references to Crypto objects and reuse some testMessages.
